### ISSUE TITLE
Storyshots testing not working with Svelte

### ISSUE THREAD

**Issue by etsiaDigital** [storybookjs/storybook#14471]
**Using Storyshots auto testing fails**

Using and following the Svelte tutorial:
https://storybook.js.org/tutorials/intro-to-storybook/svelte/en/simple-component/

Running Sample.test.js works fine.
Running storybook.test.js (created following tutorial instructions) fails with:  "Jest encountered an unexpected token"

Details: .../node_modules/@storybook/svelte/dist/cjs/client/preview/SlotDecorator.svelte:1   
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){<script>
                                                                                                                                        ^
It appears that the precompile of the svelte file isn't happening.
I tried svelte-jester as an alternative (to jest-transform-svelte) and got these same results.  

**To Reproduce**

1. Follow the Svelte tutorial.    https://storybook.js.org/tutorials/intro-to-storybook/svelte/en/simple-component/
2. After the Snapshot Testing instructions are performed, run "yarn test" as instructed and I get the error.  

**Expected behavior**
I believe this wouldn't happen if the precompile of the .svelte files occurred as expected so Jest was being fed JS as it requires.

**Comment by j3rem1e** [comment]
We can't ship precompiled js of the svelte files, because they should be compiled against the version on your project.

However, you just had to tell jest to transform files in `@storybook/svelte` : 

```js
transformIgnorePatterns: ['node_modules/(?!(@storybook/svelte)/)'],
```

**Comment by etsiaDigital** [comment]
You misunderstood me about "precompiled".  (I meant it as used here: https://github.com/mihar-22/svelte-jester ) I might have misunderstood the error as this is the first time I've used Storybook much less Storyshots.  And I'm new to Jest (I'm from the C++ world and a different workflow).  I noticed that both 'svelte-jester' and, what is shipped with Storybook: 'jest-transform-svelte', both do the same thing in this respect.

However, the "bug" is then with the Tutorial.

Adding the following to package.json in the "jest" section fixed the issue:
    "transformIgnorePatterns": ["node_modules/(?!(@storybook/svelte)/)"],
